test/svg-sprite/shape/shape.constructor.test.js

```diff
@@ -158,6 +166,28 @@ describe('testing Shape.constructor', () => {
                     expect(shape.config.id.generator(`test${path.sep}test.f.svg`)).toBe('test.f-test');
                     expect(shape.config.id.generator('test 1.svg')).toBe('test_1-test');
                 });
+
+                it('should id shape keep folder names from relative path', () => {
+                    expect.hasAssertions();
+
+                    const TEST_SPRITER = {
+                        config: {
+                            shape: {
+                                id: {
+                                    generator: true
+                                },
+                                meta: {},
+                                align: {}
+                            }
+                        },
+                        verbose: jest.fn()
+                    };
+                    const shape = new SVGShape(TEST_FILE_WITH_FOLDERS, TEST_SPRITER);
+
+                    expect(shape.config.id.generator(TEST_FILE_WITH_FOLDERS.relative)).toBe('folder--test_path.f');
+                    expect(path.relative(TEST_FILE_WITH_FOLDERS.base, TEST_FILE_WITH_FOLDERS.path)).toBe(TEST_FILE_WITH_FOLDERS.relative);
```

Comment:
This line is unnecessary because it testing the NodeJS functionality

Reply:
Changed to obtained the relative content from the content of the base path and full path.